iroh-net/src/magicsock.rs

```diff
@@ -2000,39 +2006,39 @@ impl Actor {
         out
     }
 
-    /// Refreshes knowledge about our local endpoints.
+    /// Refreshes knowledge about our direct addresses.
     ///
     /// In other words, this triggers a netcheck run.
     ///
-    /// Note that invoking this is managed by the [`EndpointUpdateState`] and this should
+    /// Note that invoking this is managed by the [`DirectAddrUpdateState`] and this should
     /// never be invoked directly.  Some day this will be refactored to not allow this easy
     /// mistake to be made.
     #[instrument(level = "debug", skip_all)]
-    async fn update_endpoints(&mut self, why: &'static str) {
-        inc!(MagicsockMetrics, update_endpoints);
+    async fn update_direct_addrs(&mut self, why: &'static str) {
+        inc!(MagicsockMetrics, update_direct_addrs);
 
-        debug!("starting endpoint update ({})", why);
+        debug!("starting direct addr update ({})", why);
         self.port_mapper.procure_mapping();
         self.update_net_info(why).await;
     }
 
-    /// Stores the results of a successful endpoint update.
-    async fn store_endpoints_update(&mut self, nr: Option<Arc<netcheck::Report>>) {
+    /// Stores the results of a successful direct addr update.
+    async fn store_direct_addr_update(&mut self, nr: Option<Arc<netcheck::Report>>) {
         let portmap_watcher = self.port_mapper.watch_external_address();
 
-        // endpoint -> how it was found
+        // direct_addr -> how it was found
         let mut already = HashMap::new();
-        // unique endpoints
-        let mut eps = Vec::new();
+        // unique direct addrs
+        let mut addrs = Vec::new();
 
         macro_rules! add_addr {
-            ($already:expr, $eps:expr, $ipp:expr, $et:expr) => {
+            ($already:expr, $addrs:expr, $ipp:expr, $typ:expr) => {
                 #[allow(clippy::map_entry)]
                 if !$already.contains_key(&$ipp) {
-                    $already.insert($ipp, $et);
-                    $eps.push(DirectAddr {
+                    $already.insert($ipp, $typ);
+                    $addrs.push(DirectAddr {
                         addr: $ipp,
-                        typ: $et,
+                        typ: $typ,
                     });
                 }
             };
```

Comment:
Not relevant to this PR, but why is this a macro, not a function?

Reply:
why not :P

Reply:
I see you half-joking, but I'll still explain myself: It's mostly about complexity. I'd argue macros are more complex and that leads to a bunch of inefficiencies. Tooling gets confused/disabled/slow more often, people are more likely to shy away from that code, etc. Same arguments as in any other case with more complexity, e.g. when there's too many abstraction layers.

Reply:
I never joke ;) 

The answer is when I wrote this originally, it was easier to use a macro, I don't remember why though

Reply:
yeah, I share @matheus23's opinion here.  I tried to re-write this once as a function and it was impossible due to borrowing rules.  It probably is still possible with a bit more refactoring though, I just recall abandoning it for some reason.